index.js

```diff
@@ -226,6 +229,21 @@ function appendObjectKeys (str, key, obj, transformer) {
   return newStr
 }
 
+function appendObjectEntries(str, key, obj, joiner) {
+  let newStr = str;
+  Array.from(Object.entries(obj)).forEach(([k, v]) => {
+    newStr = append(
+      newStr,
+      key,
+      { key: k, val: v },
+      typeof joiner === "function"
+        ? joiner
+        : (agg) => `${quote(agg.key)}${joiner}${quote(agg.val)}`
```

Comment:
Is there a particular reason to wrap keys and values in quotes separately instead of together with the `=` sign? Why not use the same format as environment variables?

e.g. `-l 'com.docker.compose.version=1.24.1'`

According to my tests locally, escaping still works correctly and `docker` still parses the key-value pairs correctly.

Reply:
I was thinking that quoting the key and value separately would allow for a case where there is equals (`=`) sign in the key or the value. But I haven't tested it, I'd gladly change it to quote the full thing if you know that this is not correct.

Reply:
According to the [docs on Docker object labels](https://docs.docker.com/config/labels-custom-metadata/), keys should only use a limited set of characters:

> Label keys should begin and end with a lower-case letter and should only contain lower-case alphanumeric characters, the period character (`.`), and the hyphen character (`-`). Consecutive periods or hyphens are not allowed.

and values can be just about any string:

> Label values can contain any data type that can be represented as a string, including (but not limited to) JSON, XML, CSV, or YAML. The only requirement is that the value be serialized to a string first, using a mechanism specific to the type of structure. For instance, to serialize JSON into a string, you might use the `JSON.stringify()` JavaScript method.

So it makes sense to quote the value, but I don't think we need to quote the key (or the equal sign), since all the characters that Docker says should be used in a key are [shell-safe](https://stackoverflow.com/questions/15783701/which-characters-need-to-be-escaped-when-using-bash/44581064#44581064).